packages/lambda-tiler/src/wmts.capability.ts

```diff
@@ -288,26 +339,64 @@ export class WmtsCapabilities {
       }),
     ]);
   }
+
   toVNode(): VNodeElement {
-    const layers: (VNodeElement | null)[] = [];
-    layers.push(this.buildLayer(this.tileSet));
-
-    if (this.layers) {
-      const layerByName = new Map<string, ConfigLayer>();
-      // Dedupe the layers by unique name
-      for (const img of this.layers) layerByName.set(standardizeLayerName(img.name), img);
-      const orderedLayers = Array.from(layerByName.values()).sort((a, b) =>
-        (a.title ?? a.name).localeCompare(b.title ?? b.name),
-      );
-      for (const img of orderedLayers) layers.push(this.buildLayerFromImagery(img));
-    }
+    // Prepare provider vNode if exists
+    const provider = this.toProviderVNode(this.provider);
+
+    // Build TileSet Layer VNodes
+    const layers: VNodeElement[] = [];
+    layers.push(this.toLayerVNode(this.tileSet));
+    const contents = layers.concat(this.toAllImageryLayersVNode(this.configLayers));
+
+    // Build TileMatrix Sets vNodes
+    for (const tms of this.tileMatrixSets.values()) contents.push(this.buildTileMatrixSet(tms));
+
+    return V('Capabilities', CapabilitiesAttrs, [...provider, V('Contents', contents)]);
+  }
+}
 
-    for (const tms of this.tileMatrixSets.values()) layers.push(this.buildTileMatrixSet(tms));
+export interface WmtsCapabilitiesParams {
+  provider?: WmtsProvider;
+  /** Tileset to export into WMTS */
+  tileSet: ConfigTileSet;
+  /** List of tile matrixes to output */
+  tileMatrix: TileMatrixSet[];
+  /** All the imagery used by the tileSet and tileMatrixes */
+  imagery: Map<string, ConfigImagery>;
+  /** Limit the output to the following image formats other wise @see ImageFormatOrder */
+  formats?: ImageFormat[];
+  /** Specific layers to add to the WMTS */
+  layers?: ConfigLayer[];
+}
 
-    return V('Capabilities', CapabilitiesAttrs, [...this.buildProvider(), V('Contents', layers)]);
+/**
+ * WMTS Capabilities Builder
+ *
+ * /v1/tiles/:tileSet/:tileMatrix/WMTSCapabilities.xml
+ * /v1/tiles/:tileSet/WMTSCapabilities.xml
+ * /v1/tiles/WMTSCapabilities.xml
+ *
+ * @example
+ *
+ */
+export class WmtsCapabilities {
+  builder: WmtsCapabilitiesBuilder;
+
+  constructor(builder: WmtsCapabilitiesBuilder, params: WmtsCapabilitiesParams) {
```

Comment:
Delete this class, rename build to `WmtsCapabilities` then this function could be `WmtsCabilities.fromParams(parms)`